rdrf/report/reports/generator.py

```diff
@@ -197,141 +153,218 @@ def export_to_json(self, request):
-
-        # Rename the columns being pivoted to occur alphabetically in the order desired in the output report
-        # https://github.com/pandas-dev/pandas/issues/17041#issuecomment-317576297
-        df.rename(inplace=True, columns={'cfg.defaultName': 'a.cfg.defaultName',
-                                         'cde.value': 'b.cde.value'})
-
-        # Pivot the cde values by their uniquely identifying columns (context form group, form, section, cde)
-        cfg_name_col, form_name_col, section_name_col, cde_name_col = get_cde_pivot_columns(self.report_design.cde_heading_format)
-
-        cde_pivot_cols = [cfg_name_col, 'cfg.sortOrder', 'cfg.entryNum', form_name_col, section_name_col, 'section.entryNum', cde_name_col]
-
-        pivoted = df.pivot(index=demographic_cols,
-                           columns=cde_pivot_cols,
-                           values=['a.cfg.defaultName', 'b.cde.value'])
-
-        # Re-order the columns
-        pivoted = pivoted.sort_index(axis=1, level=cde_pivot_cols)
-
-        # Remove context form group's sort order as we don't need to see this in the results.
-        pivoted = pivoted.droplevel('cfg.sortOrder', axis=1)
-
-        # Flatten the column levels into one, and reformat labels
-        pivoted.columns = pivoted.columns.to_series().str.join('_')
-        pivoted.columns = pivoted.columns.to_series().apply(self.__reformat_pivoted_column_labels)
-
-        # Remove null columns (caused by patients with no matching clinical data)
-        pivoted = pivoted.loc[:, pivoted.columns.notnull()]
-        # Remove duplicate columns (caused by repeated context form group's defaultName from how it's pivoted)
-        pivoted = pivoted.loc[:, ~pivoted.columns.duplicated()]
-
-        return stream_to_csv(pivoted)
+        def get_demographic_headers():
+            def get_flat_json_path(report_model, report_field, i=None):
+                if not report_field:
+                    return None
+                if report_model == 'patient':
+                    prefix = ''
+                else:
+                    prefix = f'{report_model}_'
+
+                field_json_path = re.sub(r"\s", "", report_field)
+                field_json_path = re.sub(r"(.+)({(.*)})", r"\1_\3", field_json_path)
+                if i is not None:
+                    return f"{prefix}{i}_{field_json_path}"
+                else:
+                    return f"{prefix}{field_json_path}"
+
+            def get_variants(lookup_key):
+                query = GqlQuery().fields([lookup_key]).query('dataSummary',
+                                                              input=self.patient_filters).operation().generate()
+                summary_result = self.schema.execute(query, context_value=request)
+                return summary_result.data['dataSummary'][lookup_key]
+
+            fieldnames_dict = OrderedDict()
+            fieldnames_dict['id'] = 'ID'
```

Comment:
Worth setting `auto_camelcase=False` in the schema?

https://docs.graphene-python.org/en/latest/api/#schema